libwidgets/Chrome/BreadcrumbIconList.vala

```diff
@@ -33,11 +33,13 @@ namespace Marlin.View.Chrome {
 
         private GLib.List<BreadcrumbIconInfo?> icon_info_list = null;
         private Granite.Services.IconFactory icon_factory;
-        Gtk.StyleContext context;
+        private Gtk.StyleContext context;
+        public int scale { public get; private set; }
 
         public BreadcrumbIconList (Gtk.StyleContext _context) {
             context = _context;
             icon_factory = Granite.Services.IconFactory.get_default ();
+            scale = context.get_scale ();
```

Comment:
I wouldn't cache the scale value, because it might change (the stylecontext is always updated)

Reply:
That's why it's cached, because i wasn't sure if it can change. If i load icon with 2x scaling, i have to draw it with the same scaling it was loaded with. Caching value just makes sure icon isn't drawn too small or too big.

Do you know how scaling could change while app is running? If that's possible then i would have to detect it and reload icons.

Reply:
Manually editing org.gnome.desktop.interface.scaling-factor is possible while the app is running. Or running the app under the environment GTK_DEBUG=interactive allows rescaling.